test/integration/targets/git/tasks/separate-git-dir.yml

```diff
@@ -0,0 +1,64 @@
+# test code for repositories with separate git dir updating
+# see https://github.com/ansible/ansible/pull/38016
+# see https://github.com/ansible/ansible/issues/30034
+
+- name: SEPARATE-GIT-DIR | clear checkout_dir
+  file:
+    state: absent
+    path: '{{ checkout_dir }}'
+
+- name: create a tempdir for separate git dir
+  local_action: shell mktemp -du
+  register: tempdir
+
+- name: SEPARATE-GIT-DIR | clone with a separate git dir
+  command: git clone {{ repo_format1 }} {{ checkout_dir }} --separate-git-dir={{ tempdir.stdout }}
+
+- name: SEPARATE-GIT-DIR | update repo the usual way
+  git:
+    repo: "{{ repo_format1 }}"
+    dest: "{{ checkout_dir }}"
+
+- name: SEPARATE-GIT-DIR | set git dir to non-existent dir
+  shell: "echo gitdir: /dev/null/non-existent-dir > .git"
```

Comment:
We also need to test the case where `gitdir: ...` is on the other line.

Reply:
Like this https://github.com/ansible/ansible/pull/38016/commits/82430f6c2d0c80475177cbb2dbae0857606abeeb ?

Reply:
Except it should succeed

Reply:
That's why I asked you to reuse existing code: https://github.com/ansible/ansible/pull/38016#issuecomment-387800545

Reply:
Okay

Reply:
To be clear: don't copy-paste blindly, merge the best parts of the approaches ;)